x-pack/platform/packages/shared/kbn-ai-assistant/src/chat/welcome_message_knowledge_base.tsx

```diff
@@ -40,44 +40,61 @@ export function WelcomeMessageKnowledgeBase({
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
   const handleClosePopover = () => setIsPopoverOpen(false);
 
-  const [checkForInstallStatus, setCheckForInstallStatus] = useState(false);
+  const [pollKnowledgeBaseStatus, setPollKnowledgeBaseStatus] = useState(false);
 
-  // When the knowledge base is installed, show a success message for 3 seconds
+  // Tracks whether the inference endpoint creation process has started
+  const inferenceEndpointIsInstalling = knowledgeBase.isInstalling;
+
+  // Tracks whether the model is fully ready
+  const modelIsReady = knowledgeBase.status.value?.ready === true;
+
+  // Determines if the model deployment is still in progress
+  // This happens when the model is not ready but the endpoint exists
+  const modelDeploymentInProgress = !modelIsReady && !!knowledgeBase.status.value?.endpoint;
+
+  // Determines if the overall installation process is ongoing
+  // Covers both the endpoint setup phase and the model deployment phase
+  const isInstalling = inferenceEndpointIsInstalling || modelDeploymentInProgress;
+  // start polling kb status if inference endpoint is being created or has been created but model isn't ready
+  useEffect(() => {
+    if (isInstalling) {
+      setPollKnowledgeBaseStatus(true);
+    }
+  }, [isInstalling]);
+
+  // When the knowledge base is installed and ready, show a success message for 3 seconds
   useEffect(() => {
-    if (previouslyNotInstalled && knowledgeBase.status.value?.ready) {
+    if (previouslyNotInstalled && modelIsReady) {
       setTimeoutTime(3000);
       reset();
       setShowHasBeenInstalled(true);
     }
-  }, [knowledgeBase.status.value?.ready, previouslyNotInstalled, reset]);
+  }, [modelIsReady, previouslyNotInstalled, reset]);
 
-  // When the knowledge base is installed, stop checking for install status
+  // When the knowledge base is ready, stop polling for status
   useEffect(() => {
-    if (!checkForInstallStatus && knowledgeBase.status.value?.ready) {
-      setCheckForInstallStatus(false);
+    if (modelIsReady) {
+      setPollKnowledgeBaseStatus(false);
     }
-  }, [checkForInstallStatus, knowledgeBase.status.value?.ready]);
+  }, [pollKnowledgeBaseStatus, modelIsReady]);
```

Comment:
Is `pollKnowledgeBaseStatus` needed as a dep here? It's not used in the `useEffect` hook, is it?

```suggestion
  }, [modelIsReady]);
```